serveradmin/serverdb/models.py

```diff
@@ -748,6 +758,70 @@ def __str__(self):
         return str(self.change_on)
 
 
+class Change(models.Model):
+    class Type(models.TextChoices):
+        CREATE = 'create', _('create')
+        CHANGE = 'change', _('change')
+        DELETE = 'delete', _('delete')
+
+    class ChangeJSONEncoder(DjangoJSONEncoder):
+        _NETWORK_TYPES = (
+            IPv4Address,
+            IPv6Address,
+            IPv4Network,
+            IPv6Network,
+            EUI,
+        )
+
+        # This is close to json_encode_extra used in adminapi.request but with
+        # two differences. First we don't need to take care about BaseFilter
+        # objects as they are already "resolved". Second DjangoJSONEncoder
+        # handles datetime on its own. Third we don't blindly cast everything
+        # else to string but explicitly check for the types we now and fail
+        # for the others.
+        def default(self, obj):
+            # Handles ServerInetAttribute and ServerMACAddressAttribute values
+            if isinstance(obj, self._NETWORK_TYPES):
+                return str(obj)
+            # Handles MultiAttr values
+            elif isinstance(obj, set):
+                return list(obj)
+
+            return super().default(obj)
+
+    object_id = models.IntegerField(db_index=True)
+    change_type = models.CharField(choices=Type.choices, max_length=6)
```

Comment:
I searched the docs and didn't found a Django native way to specify a PostgreSQL native enum type. 

We could of course write a custom migration but would still require this definition here so that Django knows how to render this field for example in the admin panel. 

I don't see a direct benefit aside from safety in SQLs directly executed. I'll add a todo here for later clean-up.